src/components/tabs/hooks/use-has-overflow.ts

```diff
@@ -0,0 +1,139 @@
+import { MutableRefObject, useRef, useState } from 'react'
+import useSafeLayoutEffect from 'hooks/use-safe-layout-effect'
+
+const MUTATION_OPTS = { attributes: true, childList: true, subtree: true }
+
+/**
+ * Overflow is determined by whether the width of a target element’s content is
+ * greater than the width of its layout.
+ *
+ * Overflow is checked on viewport resize and on a target element’s DOM
+ * mutation. Both are throttled by animation frame requests.
+ *
+ * MutationObserver Compatibility:
+ * 	- Chrome 26+
+ * 	- Edge 12+
+ * 	- Firefox 14+
+ * 	- Internet Explorer 11+
+ * 	- Safari 17+
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
+ */
+export default function useHasOverflow<T extends HTMLElement>(
+	ref: MutableRefObject<T>
```

Comment:
Could you share a bit more on this change to accept a ref rather than handling it in the hook?